flutter/shell/platform/tizen/nui_autofill_popup.cc

```diff
@@ -0,0 +1,90 @@
+
+#include <dali-toolkit/dali-toolkit.h>
+#include <dali-toolkit/devel-api/controls/table-view/table-view.h>
+#include <dali-toolkit/public-api/controls/text-controls/text-label.h>
+
+#include "flutter/shell/platform/tizen/tizen_autofill.h"
+
+namespace flutter {
+
+bool NuiAutofillPopup::Touched(Dali::Actor actor,
+                               const Dali::TouchEvent& event) {
+  const Dali::PointState::Type state = event.GetState(0);
+  if (Dali::PointState::DOWN == state) {
+    std::string text =
+        actor.GetProperty(Dali::Actor::Property::NAME).Get<std::string>();
+    on_commit_(text);
+    Hide();
+  }
+  return true;
+}
+
+void NuiAutofillPopup::Hide() {
+  // TODO(Swanseo0) : There is a phenomenon where white traces remain for a
+  // while when popup disappears.
+  popup_.SetDisplayState(Dali::Toolkit::Popup::HIDDEN);
+}
+
+void NuiAutofillPopup::Hidden() {
+  popup_.Unparent();
+  popup_.Reset();
+}
+
+void NuiAutofillPopup::OutsideTouched() {
+  Hide();
+}
+
+void NuiAutofillPopup::PrepareAutofill() {
+  popup_ = Dali::Toolkit::Popup::New();
+  popup_.SetProperty(Dali::Actor::Property::NAME, "popup");
+  popup_.SetProperty(Dali::Actor::Property::PARENT_ORIGIN,
+                     Dali::ParentOrigin::TOP_LEFT);
+  popup_.SetProperty(Dali::Actor::Property::ANCHOR_POINT,
+                     Dali::AnchorPoint::TOP_LEFT);
+  popup_.SetProperty(Dali::Toolkit::Popup::Property::TAIL_VISIBILITY, false);
+  popup_.SetBackgroundColor(Dali::Color::WHITE_SMOKE);
+  popup_.OutsideTouchedSignal().Connect(this,
+                                        &NuiAutofillPopup::OutsideTouched);
+  popup_.HiddenSignal().Connect(this, &NuiAutofillPopup::Hidden);
+  popup_.SetProperty(Dali::Toolkit::Popup::Property::BACKING_ENABLED, false);
+  popup_.SetProperty(Dali::Toolkit::Popup::Property::AUTO_HIDE_DELAY, 2500);
+}
+
+void NuiAutofillPopup::PopupAutofill(Dali::Actor* actor) {
+  const std::vector<std::unique_ptr<AutofillItem>>& items =
+      TizenAutofill::GetInstance().GetAutofillItems();
+  if (items.size() > 0) {
+    PrepareAutofill();
+    Dali::Toolkit::TableView content =
+        Dali::Toolkit::TableView::New(items.size(), 1);
+    content.SetResizePolicy(Dali::ResizePolicy::FILL_TO_PARENT,
+                            Dali::Dimension::ALL_DIMENSIONS);
+    content.SetProperty(Dali::Actor::Property::PADDING,
+                        Dali::Vector4(10, 10, 0, 0));
+    for (uint32_t i = 0; i < items.size(); ++i) {
+      Dali::Toolkit::TextLabel label =
+          Dali::Toolkit::TextLabel::New(items[i]->label_);
+      label.SetProperty(Dali::Actor::Property::NAME, items[i]->value_);
+      label.SetResizePolicy(Dali::ResizePolicy::DIMENSION_DEPENDENCY,
+                            Dali::Dimension::HEIGHT);
+      label.SetProperty(Dali::Toolkit::TextLabel::Property::TEXT_COLOR,
+                        Dali::Color::WHITE_SMOKE);
+      label.SetProperty(Dali::Toolkit::TextLabel::Property::POINT_SIZE, 7.0f);
```

Comment:
Q. Are there any default values for these properties? It'd be nice if we could avoid using magic numbers.

Reply:
The default `text color` is black and the `point size` is about 15.
The background color of the pop-up window is dark blue, so I can't see the letters well, so I set it to white smoke.
The font size is set to an appropriate size because the default value is large, but we will modify it so that we do not use the magic number.
However, if we do not change the background color of popup, white smoke will be better than the basic color(black).

Reply:
Does "white background" and "7.0f font size" look good for both tv and common (rpi) profiles?

 +) NUI popup may differ from Dali popup. It might be better to set it equal to that value.
https://docs.tizen.org/application/dotnet/guides/user-interface/nui/nui-components/Popup/